### ISSUE TITLE
Suggestion: add editor parameters for truck, bus and other vehicles

### ISSUE THREAD

**Issue by JumboSK** [osmandapp/Osmand#6729]
Hello

It will be nice, if will be posible to edit parameters for other vehicles.
basic parameters what we need:
max speed edit for specific roads like highway, basic road... (in my country truck has 90km/h , bus 100km/h on highway)
lenght (this is in menu but its only big jump between values)
height (this is in menu but its only 0,5m between values)
access forbidden (some areas are restricted for truck or bus)

sorry 1.time here, i try create and edit routing.xml but it not works :/

**Comment by aceman444** [comment]
Yes that would be great.

**Comment by vshcherb** [comment]
We are going to support custom routing .xml and custom profiles in 3.4. Probably we could make more parameters for car-like navigation, so it will be easier to configure for trucks.

**Comment by JumboSK** [comment]
vshcherb if you do it easy to use and editable, many users buy this app. i know lot of truckers and bus drivers with IGO or Sygic truck but they have only becouse havent other option. this navigation have lot of good things and i use it in my bus if you add this option for bus and truck drivers. maps allready had this informations.

**Comment by vshcherb** [comment]
@JumboSK we need first of all to make it easy to test for OSM-truck-drivers cause OSM has lots of missing tags. 
As of today we have Weight / Height limit but it is not enough. OsmAnd is heavily dependent on OSM-data, so it won't be easy journey but I guess we need to start step by step.

**Comment by aceman444** [comment]
Yes, the profiles for truck/bus/car can be similar in many aspects (e.g. road types), but will differ in some small areas, like maxweight limits, barriers/obstacles, maybe also default assumed maxspeed on roads (e.g. a bus or truck (with trailer) can't go 130kph as cars can). So this needs good thought at design time.

Also, as always, there are per-country differences, which isn't supported in OsmAnd in any way.

**Comment by wehkah** [comment]
> We are going to support custom routing .xml and custom profiles in 3.4. Probably we could make more parameters for car-like navigation, so it will be easier to configure for trucks.

Would it be possible to include a setting for dangerous goods (ADR) as well? The ADR rules do not only apply to trucks, but also to smaller vehicles, and they include tunnel codes, that forbid vehicles carrying dangerous goods to drive through tunnels, that are marked with a certain code. 

See here for details: https://de.wikipedia.org/wiki/Tunnelbeschr%C3%A4nkungscode

(Sorry for the German page, but I can't find that page in English!)

**Comment by aceman444** [comment]
How is "dangerous goods" restrictions tagged in OSM data? What are the tags/values for this that are marked on the roads where there are forbidden? Or do you mean 'hazmat' tags: https://wiki.openstreetmap.org/wiki/Key:hazmat ?

**Comment by wehkah** [comment]
Yes, I was talking about hazardous goods/materials. I did not know that Osmand already uses a tag for theses goods. 

But as you can see [here](https://de.wikipedia.org/wiki/Europ%C3%A4isches_%C3%9Cbereinkommen_%C3%BCber_die_internationale_Bef%C3%B6rderung_gef%C3%A4hrlicher_G%C3%BCter_auf_der_Stra%C3%9Fe) (or [here](https://en.wikipedia.org/wiki/ADR_(treaty)) in Englisch), the ADR rules are a bit complicated. There are lots of different types (gas, fluid, solid) and different classes (acid, explosive, poison, radio active, biological dangerous, etc) of hazardous goods, and it also makes a big difference, whether fluids are transported in a tank or as packed goods. 

But I would say that the most important information for drivers, no matter if car or truck, is the tunnel restriction code. A forbidden road is far more easy to avoid, because when you see the restriction sign, you will always have an opportunity to take a detour. But the tunnel restriction signs sometimes are placed long after the last possibility to take another road. So, when you reach the tunnel, you have the choice to either ignore the sign, or to stop, cause a traffic jam and call the police. Neither possibilities will make a truck driver's day a happy one.

Next to the vehicle's hight and weight, the possibility to set "hazardous goods", is the most important one, and because this information also applies to private cars, it should be choosable in car mode as well.

**Comment by aceman444** [comment]
I doubt this information is encoded in the OSM database yet. So you need to get that done first, which is outside of what OsmAnd can do. Only then can OsmAnd use that information and avoid those roads. Even then OsmAnd probably can't support all those values as they are really specific and for very few users (it can't even yet properly support country differences for the much more universal car/bicycle routing). But once the data is in OSM, it is easy for you to make a custom routing.xml file to avoid roads/tunnels of a particular hazmat restriction.

**Comment by wehkah** [comment]
I agree that this is quite a complicated matter for Osmand to include all road information needed. But at least the "hazmat" tag should be taken account of, it should be selectable as part of the vehicle settings. That would already cover a great deal of all the trouble that come with hazardous goods. And, like I said before, the regulations concerning hazardous goods do not only apply to trucks, but also to car's and little vans. And although it sounds funny: the rules for hazardous goods do even apply to bicycles! I'm not kidding you, but if you tow a barrel of acid with your bike, the hole set of ADR rules and regulations apply to you! (E.G. water reserves.) 

Although I am not sure, if they still apply to you as a pedestrian when you push your bike. ;)

**Comment by JumboSK** [comment]
can you please add slider to set weight and height? 0,5m difference is too mutch and useless.

**Comment by geedle** [comment]
I'll second that - a more granular way of inputting numbers is a must.

Also, I find the phrase "specify permitted vehicle weight limit on routes" confusing.  It is correct, but I always have to think carefully about what it means.  "Vehicle weight" or "Calculate route for vehicle weight..." would be a little more use friendly.

**Comment by aceman444** [comment]
@vscherb, that is not the same. If you set vehicle weight of 3.5t, will OsmAnd avoid roads with hgv=no ? I doubt that. The same with bus. There needs to be more checks for access restrictions on roads, than what is currently in routing.xml file for the 'car' profile.

**Comment by vshcherb** [comment]
nope they won't avoid, so please create a separate issue, we might introduce hgv check after certain threshold is hit.
This issue is quite generic and it could be treated as fixed because there is possibility to add custom routing.xml
